lib/synced/strategies/full.rb

```diff
@@ -0,0 +1,229 @@
+                local_object.attributes = default_attributes_mapping(remote)
+                local_object.attributes = local_attributes_mapping(remote)
+                if @globalized_attributes.present?
+                  local_object.attributes = globalized_attributes_mapping(remote,
+                    local_object.translations.translated_locales)
+                end
+                local_object.save! if local_object.changed?
+                local_object.tap do |local_object|
+                  synchronize_associations(remote, local_object)
+                end
+              end
+            end
+          end
+        end
+      end
+
+      private
+
+      def synchronize_associations(remote, local_object)
+        @associations.each do |association|
+          klass = association.to_s.classify.constantize
+          klass.synchronize(remote: remote[association], scope: local_object, remove: @remove)
+        end
+      end
+
+      def local_attributes_mapping(remote)
+        Hash[@local_attributes.map do |k, v|
+          [k, v.respond_to?(:call) ? v.call(remote) : remote.send(v)]
+        end]
+      end
+
+      def default_attributes_mapping(remote)
+        {}.tap do |attributes|
+          attributes[@id_key] = remote.id
+          attributes[@data_key] = remote if @data_key
+        end
+      end
+
+      def globalized_attributes_mapping(remote, used_locales)
+        empty = Hash[used_locales.map { |locale| [locale.to_s, nil] }]
+        {}.tap do |attributes|
+          @globalized_attributes.each do |local_attr, remote_attr|
+            translations = empty.merge(remote.send(remote_attr) || {})
+            attributes["#{local_attr}_translations"] = translations
+          end
+        end
+      end
+
+      # Returns relation within which local objects are created/edited and removed
+      # If no scope is provided, the relation_scope will be class on which
+      # .synchronize method is called.
+      # If scope is provided, like: account, then relation_scope will be a relation
+      # account.rentals (given we run .synchronize on Rental class)
+      #
+      # @return [ActiveRecord::Relation|Class]
+      def relation_scope
+        if @scope
+          @model_class.unscoped { @scope.send(resource_name).scope }
+        else
+          @model_class.unscoped
+        end
+      end
+
+      # Returns api client from the closest possible source.
+      #
+      # @raise [BookingSync::API::Unauthorized] - On unauthorized user
+      # @return [BookingSync::API::Client] BookingSync API client
+      def api
+        return @api if @api
+        closest = [@scope, @scope.class, @model_class].find do |object|
+                    object.respond_to?(:api)
+                  end
+        closest.try(:api) || raise(MissingAPIClient.new(@scope, @model_class))
+      end
+
+      def local_object_by_remote_id(remote_id)
+        local_objects.find { |l| l.public_send(@id_key) == remote_id }
```

Comment:
I'm actually not convinced here, if I would made it any better using `local_object`, it's a one lines you see that a single object will be a `local_object`